tests/pkg/fault-trigger/manager/vm.go

```diff
@@ -0,0 +1,205 @@
+)
+
+// ListVMs lists vms
+func (m *Manager) ListVMs() ([]*VM, error) {
+	shell := fmt.Sprintf("virsh list --all")
+	cmd := exec.Command("/bin/sh", "-c", shell)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	vms := parserVMs(string(output))
+	//"virsh domifaddr vm1 --interface eth0 --source agent"
+	for _, vm := range vms {
+		vmIP, err := getVMIP(vm.Name)
+		if err != nil {
+			glog.Errorf("can not get vm %s ip", vm.Name)
+			continue
+		}
+		vm.IP = vmIP
+	}
+	return vms, nil
+}
+
+// StopVM stops vm
+func (m *Manager) StopVM(v *VM) (string, error) {
+	shell := fmt.Sprintf("virsh destroy %s", v.Name)
+	cmd := exec.Command("/bin/sh", "-c", shell)
+	output, err := cmd.CombinedOutput()
+	return string(output), err
+}
+
+// StartVM starts vm
+func (m *Manager) StartVM(v *VM) (string, error) {
+	shell := fmt.Sprintf("virsh start %s", v.Name)
+	cmd := exec.Command("/bin/sh", "-c", shell)
+	output, err := cmd.CombinedOutput()
+	return string(output), err
+}
+
+func getVMIP(name string) (string, error) {
+	shell := fmt.Sprintf("virsh domifaddr %s --interface eth0 --source agent", name)
+	cmd := exec.Command("/bin/sh", "-c", shell)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		mac, err := getVMMac(name)
+		if err != nil {
+			return "", err
+		}
+
+		ipNeighShell := fmt.Sprintf("ip neigh | grep -i %s", mac)
+		cmd = exec.Command("/bin/sh", "-c", ipNeighShell)
+		ipNeighOutput, err := cmd.CombinedOutput()
+		if err != nil {
+			return "", err
+		}
+
+		return parserIPFromIPNeign(string(ipNeighOutput))
+	}
+
+	return parserIP(string(output))
+}
+
+func getVMMac(name string) (string, error) {
+	shell := fmt.Sprintf("virsh domiflist %s", name)
+	cmd := exec.Command("/bin/sh", "-c", shell)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	return parserMac(string(output))
+}
+
+// example input :
+// Interface  Type       Source     Model       MAC
+// -------------------------------------------------------
+// vnet1      bridge     br0        virtio      52:54:00:d4:9e:bb
+// output: 52:54:00:d4:9e:bb, nil
+func parserMac(data string) (string, error) {
+	data = stripEmpty(data)
+	lines := strings.Split(data, "\n")
+
+	for _, line := range lines {
+		if !strings.Contains(line, "bridge") {
+			continue
+		}
+
+		fields := strings.Split(line, " ")
+		if len(fields) < 5 {
+			continue
+		}
+
+		return fields[4], nil
+	}
+
+	return "", errors.New("mac not found")
+}
+
+// example input:
+// Name       MAC address          Protocol     Address
+// -------------------------------------------------------------------------------
+// eth0       52:54:00:4c:5b:c0    ipv4         172.16.30.216/24
+// -          -                    ipv6         fe80::5054:ff:fe4c:5bc0/64
+// output: 172.16.30.216, nil
+func parserIP(data string) (string, error) {
+	data = stripEmpty(data)
+	lines := strings.Split(data, "\n")
+	for _, line := range lines {
+		if !strings.HasPrefix(line, "eth0") {
```

Comment:
the interface name maybe not `eth0`.

Reply:
how about checking `ipv4` ?

Reply:
check the `MAC address` is exist is more reasonable

Reply:
https://github.com/pingcap/tidb-operator/pull/312/files#diff-cc0457c3ced0a4de0bcef6986b537b8aR72  
if this check was failed, the program will check the `mac address`